flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/AbstractTtlState.java

```diff
@@ -82,5 +88,19 @@ public void setCurrentNamespace(N namespace) {
 	@Override
 	public void clear() {
 		original.clear();
+		accessCallback.run();
+	}
+
+	/**
+	 * Check if state has expired or not and update it if it has partially expired.
+	 *
+	 * @return either non expired (possibly updated) state or null if the state has expired.
+	 */
+	@Nullable
+	public abstract TTLSV checkIfExpiredOrUpdate(@Nonnull TTLSV ttlValue);
```

Comment:
This method name sounds still like it should return a boolean as it did in the beginning, but now it does more so I would consider adjusting the name as well.